### PR TITLE
Updates to image optimization settings

### PR DESCRIPTION
**Pull request by SiobhyB**
> **Note**
> Although this PR is large, the vast majority of file changes are the result of a change to a string name, as can be viewed in https://github.com/wordpress-mobile/WordPress-Android/pull/19581/commits/452ed672ac3d147571376dc954eb907bb4f34c23

## Description

This PR changes the default setting for `Optimize Images` (found via _Me_ → _App Settings_) from off → on. We will also inform users of this new default via a popup that will display when uploading media.

The default `Maximum Image Size` has also been updated from _3000px_  to _2000px_ and the `Image Quality`'s _Medium_ value has been changed from _85%_ to _80%_. 

You can find the iOS equivalent at https://github.com/wordpress-mobile/WordPress-iOS/pull/21981.

### Included PRs

The following three PRs were merged into this one, with all of them having been separately reviewed and approved:

* https://github.com/wordpress-mobile/WordPress-Android/pull/19556
* https://github.com/wordpress-mobile/WordPress-Android/pull/19648
* https://github.com/wordpress-mobile/WordPress-Android/pull/19644

## To test

<details>
 <summary>Verify default setting displays as expected ⤵️</summary>

### Default setting should be changed if it has never been set before

* Perform a fresh installation of the app.
* Navigate to _Me → App Settings_. 
* Observe that _Optimize Images_ setting is enabled by default.
* Observe that _Maximum Image Size_ and _Image Quality_ settings have the default values as shared in the PR's description.

#### Set original as maximum image size
* Install an older version.
* Navigate to _Me → App Settings_. 
* Move the _Maximum Image Size_ setting to a different value than the default.
* Move the _Maximum Image Size_ setting back to the original value.
* Install the installable build from this PR.
* Navigate to _Me → App Settings_. 
* Observe that the _Optimize Images_ setting is enabled and that _Maximum Image Size_ setting remains in the previously set original value.

#### Set maximum image size
* Install an older version.
* Navigate to _Me → App Settings_. 
* Move the _Maximum Image Size_ setting to a different value than the original.
* Install the installable build from this PR.
* Navigate to _Me → App Settings_. 
* Observe that the _Optimize Images_ setting is enabled and that _Maximum Image Size_ setting remains in the previously set value.
</details>

<details>
 <summary>Resize settings should function as expected ⤵️</summary>

* Add an image to the device with dimensions above 2000px.
* Remove and install the app.
* Navigate to the Posts/Pages screen.
* Open/create a post/page.
* Add an Image block.
* Tap on the _Choose from device_ option.
* Select the image previously added with large dimensions.
* When the _Optimize Images_ pop-up is displayed, select the option to keep image optimization.
* Wait for the image upload to finish.
* Go to the browser and navigate to the Media screen of the site.
* Select the uploaded image and click on the _Edit_ button.
* Observe that dimensions displayed under the _DIMENSIONS_ section are lower than the original ones. The dimensions shouldn’t exceed the value set in _Max Image Upload Size_ setting.
* Observe that the file size under the _FILE SIZE_ section is smaller than the original file size.</details>

<details>
 <summary>Compression settings should function as expected ⤵️</summary>

- Add an image to the device with a large file size.
- Remove and install the app.
- Navigate to the Posts/Pages screen.
- Open/create a post/page.
- Add an Image block.
- Tap on the _Choose from device_ option.
- Select the image previously added with a large file size.
- When the _Optimize Images_ pop-up is displayed, select the option to keep image optimization.
- Wait for the image upload to finish.
- Go to the browser and navigate to the Media screen of the site.
- Select the uploaded image and click on the _Edit_ button.
- Observe that the file size under the _FILE SIZE_ section is smaller than the original file size.</details>

<details>
<summary>Pop-up should be displayed in specific sources ⤵️</summary>

### Media Screen
- Remove and install the app.
- Navigate to `My Site → Media`.
- Tap on ➕ button and select `Choose from My Device` option.
- Select one or multiple items.
- Tap on the `Add` button.
- Observe that the Optimize Images pop-up is displayed.

### Gutenberg Editor

**Preparation:**
- Remove and install the app.
- Navigate to `My Site → Posts` (or Pages).
- Create a post.

#### Media Blocks - From Device
- Add any of the following blocks:
  - Image block
  - Gallery block
  - Cover block
  - Media & Text block
  - Story block.
- Tap on the `Choose from device` option.
- Select one item (or multiple items when using Gallery block).
- Observe that the Optimize Images pop-up is displayed.

#### Media Blocks - Take Photo

> **Note**
> The pop up doesn't currently display when taking a photo from the Story block. This will be worked on in a future PR.

- Add any of the following blocks:
  - Image block
  - Gallery block
  - Cover block
  - Media & Text block
- Tap on the `Take a Photo` option.
- Take a photo and use it.
- Observe that the Optimize Images pop-up is displayed.

### Featured Image - From Device
- Tap on the `...` button located on the top bar.
- Tap on the `Post Settings` options.
- Navigate to the Featured Image section and tap on `Set Featured Image`.
- Tap on the `Choose from Device` option.
- Select one time.
- Observe that the Optimize Images pop-up is displayed.

### Featured Image - Take Photo
- Tap on the `...` button located on the top bar.
- Tap on the `Post Settings` options.
- Navigate to the Featured Image section and tap on `Set Featured Image`.
- Tap on the `Take a Photo` option.
- Select one time.
- Observe that the Optimize Images pop-up is displayed.</details>

<details>
 <summary>Check events ⤵️</summary>

Verify that the new `APP_SETTINGS_OPTIMIZE_IMAGES_POPUP_TAPPED` event is sent when tapping either of the options in the new popup. `on` will be sent if you leave optimization on and `off` if it's turned off via the popup.
</details>

## Regression Notes
1. Potential unintended areas of impact

As we're changing default settings, there's a risk of unintentionally changing settings users have chosen. We're also changing code around the App Settings screen and media upload flows, so there could be unintended impact there.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manually tested each flow we identified and relied on existing tests around media uploads.

3. What automated tests I added (or what prevented me from doing so)

I plan to further analyse existing tests and potentially add new ones in a future PR.

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values-ar/strings.xml
Comment (fluiddot) [inline WordPress/src/main/res/values-ar/strings.xml:3003]: I wonder if we had to update all localizations. Based on the file history ([example](https://github.com/wordpress-mobile/WordPress-Android/commit/6d575e4ea4964358ec91ee94708145b37ba00d28)), seems these localization files are updated automatically as part of the release process.

File: WordPress/src/main/res/values/strings.xml
Comment (fluiddot) [inline WordPress/src/main/res/values/strings.xml]: I noticed one of the break lines (`\n`) only contains `\` after the change:

```suggestion
    <string name="image_optimization_promo_desc">Image optimization shrinks images for faster uploading.\n\nThis option is enabled by default, but you can change it in the app settings at any time.</string>
```

File: WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java
Comment (fluiddot) [inline WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java]: Nitpick: These two conditions could be grouped into a single `if` statement.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! Awesome work @SiobhyB 🏅 !

I've tested the different scenarios described in the PR's description and confirmed the changes behave as expected ✅ . I added some comments, [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/19581#discussion_r1403519469) in specific would be great to address it before merging the PR.

**Comment by fluiddot** [comment]
@SiobhyB As a side note and regarding the test section `Default setting should not be changed if it has been set previously`:

Since we've changed the quality values, if the Low value was previously set, the quality displayed after upgrading the app will be Medium. Not sure if users might be confused with this change, I presume that not many users use the lowest quality, hence probably it would be a minor percentage.

File: WordPress/src/main/res/values-ar/strings.xml
Comment (SiobhyB) [reply WordPress/src/main/res/values-ar/strings.xml:3003]: Ah, good to know for the future!

File: WordPress/src/main/res/values/strings.xml
Comment (SiobhyB) [reply WordPress/src/main/res/values/strings.xml]: Committed that change, thanks for spotting!

File: WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java
Comment (SiobhyB) [reply WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java]: Refactored in https://github.com/wordpress-mobile/WordPress-Android/pull/19581/commits/51ca8afa792ac79662aedf7339ef56280990564a. 🙇‍♀️

**Comment by SiobhyB** [comment]
Thanks @fluiddot! 🙌 

> Since we've changed the quality values, if the Low value was previously set, the quality displayed after upgrading the app will be Medium. Not sure if users might be confused with this change, I presume that not many users use the lowest quality, hence probably it would be a minor percentage.

I agree it may confuse a small percentage of people but, as they'll keep the 80% compression rate, hopefully most won't notice. 🤞 I'll go ahead to merge this PR once the tests pass.

**Pull request merged**
